### PR TITLE
Fix bug with multiple TTL merges on expired column

### PR DESCRIPTION
**Pull request by alesapin**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix rare bug with column TTL for MergeTree engines family: In case of repeated vertical merge the error `Cannot unlink file ColumnName.bin ... No such file or directory.` could happen.

### CODE REVIEW

File: src/Storages/MergeTree/IMergedBlockOutputStream.cpp [hunk 1]
```diff
@@ -79,10 +79,17 @@ NameSet IMergedBlockOutputStream::removeEmptyColumnsFromPart(
...
 
     /// Remove files on disk and checksums
-    for (const String & removed_file : remove_files)
+    for (auto itr = remove_files.begin(); itr != remove_files.end();)
     {
-        if (checksums.files.contains(removed_file))
-            checksums.files.erase(removed_file);
+        if (checksums.files.contains(*itr))
+        {
+            checksums.files.erase(*itr);
```
Comment (antonio2368) [inline src/Storages/MergeTree/IMergedBlockOutputStream.cpp:86]:
```suggestion
            checksums.files.erase(*itr);
            ++itr;
```
?

Comment (alesapin) [reply src/Storages/MergeTree/IMergedBlockOutputStream.cpp:86]: Thank you, changed this code 10 times to reproduce the bug with test and finally forget normal case :)

**Review by antonio2368 — APPROVED** [review]

**Comment by alesapin** [comment]
My test is flaky.

**Comment by alesapin** [comment]
Fuzzer failures related to memory consumption: `DB::Exception: Memory limit (total) exceeded: would use 57.75 GiB 
(attempt to allocate chunk of 4204713 bytes), maximum: 55.82 GiB. OvercommitTracker decision: Memory overcommit isn't used. OvercommitTracker isn't set.`

**Pull request merged**